src/lib/pytch/syscalls.js

```diff
@@ -3,6 +3,69 @@ var $builtinmodule = function (name) {
 
     const skulpt_function = Sk.pytchsupport.skulpt_function;
 
+    const validKeys = ["a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", 
+    "l", "m", "n", "o", "p", "q", "r", "s", "t", "u", "v", "w", "x", "y", "z", 
+    "0", "1", "2", "3", "4", "5", "6", "7", "8", "9",
+    " ", "ArrowLeft", "ArrowDown", "ArrowUp", "ArrowRight"];
+
+    const map = {};
+
+    validKeys.forEach(validKey => {
+        if (validKey !== " ")
+            map[validKey.toLowerCase()] = validKey;
+    });
+
+    const isValidKeyname = (keyname) => validKeys.includes(keyname)
+    
+    function suggestedKeyname(invalidKeyname) {
+        //Remove symbols and capital letters to suggest a keyname
+        //Eg "Arrowleft", "arrowLeft" or "Arrow-Left" -> "ArrowLeft"
+        const invalidKeynameLower = invalidKeyname.toLowerCase();
+        var invalidKeynameCleaned = invalidKeynameLower.trim().replace(/[^a-z0-9]/g, "");
```

Comment:
The `invalidKeynameCleaned` variable can be `const`.  And the `replace()` will get rid of whitespace (including newlines), so the `trim()` isn't needed.